platforms/core-execution/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorIntegrationTest.groovy

```diff
@@ -915,4 +915,135 @@ class WorkerExecutorIntegrationTest extends AbstractWorkerExecutorIntegrationTes
 
         fixture.addImportToBuildScript("org.gradle.test.TestWorkAction")
     }
+
+    def "classLoaderIsolation doesn't cause java.lang.OutOfMemoryError: Metaspace"() {
```

Comment:
This test is messy, because I'm not familiar with Groovy or the Gradle test tooling. If I have made mistakes, please commit fixes directly to the branch.